File: iOS/Objective C/IronSourceDemoApp/InitializationDelegate.m
//
//  InitializationDelegate.m
//  IronSourceDemoApp
//
//  Copyright © 2024 ironSource Mobile Ltd. All rights reserved.
//

#import "InitializationDelegate.h"

@implementation InitializationDelegate

- (instancetype)initWithDelegate:(id<DemoViewControllerDelegate>)delegate {
    self = [super init];
    
    if (self) {
        _delegate = delegate;
    }
    
    return self;
}

/**
 Called after the Mediation successfully complets its initialization
 */
- (void)initializationDidComplete {
    logCallbackName();
    
    [self.delegate setEnablementForButton:LoadInterstitialButtonIdentifier
                                   enable:YES];
    [self.delegate setEnablementForButton:LoadBannerButtonIdentifier
                                   enable:YES];
}

@end

File: iOS/Swift/IronSourceSwiftDemoApp/InitializationDelegate.swift
//
//  InitializationDelegate.swift
//  IronSourceSwiftDemoApp
//
//  Copyright © 2024 ironSource Mobile Ltd. All rights reserved.
//

import Foundation
import IronSource

class InitializationDelegate: NSObject, ISInitializationDelegate {
    
    weak var delegate: DemoViewControllerDelegate?

    init(delegate: DemoViewControllerDelegate!) {
        self.delegate = delegate
    }
    
    /**
     Called after the Mediation successfully complets its initialization
     */
    func initializationDidComplete() {
        print("InitializationDelegate \(#function)")
        
        self.delegate?.setButtonEnablement(ButtonIdentifiers.loadInterstitialButtonIdentifier, enable: true)
        self.delegate?.setButtonEnablement(ButtonIdentifiers.loadBannerButtonIdentifier, enable: true)
    }
}

